hs_core/views/__init__.py

```diff
@@ -677,5 +677,8 @@ def _set_resource_sharing_status(request, user, resource, is_public):
         if user.uaccess.can_change_resource_flags(resource):
             resource.raccess.public = is_public
             resource.raccess.save()
+            # set isPublic metadata AVU isPublic accordingly
+            istorage = IrodsStorage()
+            istorage.setAVU(resource.short_id, "isPublic", str(is_public))
```

Comment:
I don't think this should be done in the back. If this needs done any time `.raccess.public` changes, then updating that flag, making the update to iRODS, and saving the `ResourceAccess` should all probably be in a helper method on `raccess`. Also, this needs some error handling. What happens if the `setAVU()` call fails?